flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/SubtaskCheckpointCoordinatorImpl.java

```diff
@@ -75,6 +75,8 @@
             LoggerFactory.getLogger(SubtaskCheckpointCoordinatorImpl.class);
     private static final int DEFAULT_MAX_RECORD_ABORTED_CHECKPOINTS = 128;
 
+    private static final int CHECKPOINT_EXECUTION_DELAY_LOG_THRESHOLD_MS = 30_000;
```

Comment:
Should it be connected somehow with failure timeout(or something similar I don't know the exact name)?

Reply:
Exceeding this threshold is not a failure but rather a potential problem that the user should be warned of.
So I think existing timeouts aren't related to this one.